plugins/inputs/snmp/snmp.go

```diff
@@ -620,6 +620,9 @@ func (s *Snmp) getConnection(idx int) (snmpConnection, error) {
 	var err error
 	var gs snmp.GosnmpWrapper
 	gs, err = snmp.NewWrapper(s.ClientConfig)
+	if strings.ContainsAny(agent, "udp") {
+		gs.UseUnconnectedUDPSocket = true
+	}
```

Comment:
Is this `if` needed? Couldn't we set the value anyway?

Reply:
I would rather be explicit about setting this option so it is only set when using udp to decrease confusion for possibly affecting TCP down the road. Without sending the agent to the NewWrapper there is no way to check if the user is using udp or TCP.

Reply:
I think we had such a discussion already earlier. But I still can't see how a UDP setting might ever affect anything that has to do with TCP? If that option would have named something like `UseUnconnectedSocket`, then I would agree and leave it like it is, but now it specifically mentions in the name it is only relevant for UDP..

Reply:
We have had this discussion. I want to be explicit when setting this option for easier maintenance down the road.

Reply:
Ok, I agree. But I would place this inside the `GosnmpWrapper` after this:
https://github.com/influxdata/telegraf/blob/2ca5cfed94cef0b987ce487600c631e8583aa05e/internal/snmp/wrapper.go#L145-L156

Reply:
What do you think Mya?

Reply:
moved